File: apply_constants.py
import os
import re

constants_dir = r"c:\socialsteps\socialsteps-master\lib\utils"
os.makedirs(constants_dir, exist_ok=True)

constants_content = """class AppConstants {
  // Firestore Collections
  static const String parentsCollection = "parents";
  static const String childrenCollection = "children";
  
  // Modules
  static const String moduleEmotion = "emotion";
  static const String moduleFocus = "focus";
  static const String modulePuzzle = "puzzle";
  static const String moduleColor = "color";

  // Difficulty Levels
  static const String levelEasy = "easy";
  static const String levelMedium = "medium";
  static const String levelHard = "hard";
}
"""
with open(os.path.join(constants_dir, "constants.dart"), "w", encoding="utf-8") as f:
    f.write(constants_content)

root_dir = r"c:\socialsteps\socialsteps-master\lib"
for dp, dn, filenames in os.walk(root_dir):
    for f in filenames:
        if f.endswith(".dart") and f != "constants.dart":
            path = os.path.join(dp, f)
            with open(path, "r", encoding="utf-8") as file:
                content = file.read()
            
            # Replace collections
            new_content = re.sub(r'collection\("parents"\)', r'collection(AppConstants.parentsCollection)', content)
            new_content = re.sub(r"collection\('parents'\)", r'collection(AppConstants.parentsCollection)', new_content)
            
            new_content = re.sub(r'collection\("children"\)', r'collection(AppConstants.childrenCollection)', new_content)
            new_content = re.sub(r"collection\('children'\)", r'collection(AppConstants.childrenCollection)', new_content)
            
            if new_content != content:
                import_stmt = "import 'package:socialsteps/utils/constants.dart';"
                if import_stmt not in new_content:
                    # insert after first import safely
                    new_content = re.sub(r"^(import .*;\r?\n)", r"\1" + import_stmt + "\n", new_content, count=1)
                    
                with open(path, "w", encoding="utf-8") as file:
                    file.write(new_content)

print("Constants successfully extracted and applied.")

File: apply_constants.dart
import 'dart:io';

void main() {
  final constantsDir = Directory('lib/utils');
  if (!constantsDir.existsSync()) {
    constantsDir.createSync(recursive: true);
  }

  final constantsFile = File('lib/utils/constants.dart');
  constantsFile.writeAsStringSync('''class AppConstants {
  // Firestore Collections
  static const String parentsCollection = "parents";
  static const String childrenCollection = "children";
  
  // Modules
  static const String moduleEmotion = "emotion";
  static const String moduleFocus = "focus";
  static const String modulePuzzle = "puzzle";
  static const String moduleColor = "color";

  // Difficulty Levels
  static const String levelEasy = "easy";
  static const String levelMedium = "medium";
  static const String levelHard = "hard";
}
''');

  final rootDir = Directory('lib');
  final files = rootDir.listSync(recursive: true).whereType<File>().where((file) => file.path.endsWith('.dart') && !file.path.endsWith('constants.dart'));

  for (final file in files) {
    String content = file.readAsStringSync();
    String newContent = content;

    newContent = newContent.replaceAll('collection("parents")', 'collection(AppConstants.parentsCollection)');
    newContent = newContent.replaceAll("collection('parents')", 'collection(AppConstants.parentsCollection)');

    newContent = newContent.replaceAll('collection("children")', 'collection(AppConstants.childrenCollection)');
    newContent = newContent.replaceAll("collection('children')", 'collection(AppConstants.childrenCollection)');

    if (newContent != content) {
      if (!newContent.contains("import 'package:socialsteps/utils/constants.dart';") && 
          !newContent.contains("import '../utils/constants.dart';")) {
        newContent = newContent.replaceFirst(RegExp(r'(import .*;\n)'), "\\1import 'package:socialsteps/utils/constants.dart';\\n");
      }
      file.writeAsStringSync(newContent);
    }
  }

  print("Constants successfully extracted and applied using Dart.");
}
